packages/sync-service/lib/electric/shapes/filter.ex

```diff
@@ -41,6 +51,8 @@ defmodule Electric.Shapes.Filter do
   """
   @spec add_shape(Filter.t(), shape_id(), Shape.t()) :: Filter.t()
   def add_shape(%Filter{} = filter, shape_id, shape) do
+    if is_map_key(filter.shapes, shape_id), do: raise("duplicate shape #{shape_id}")
```

Comment:
```suggestion
    if has_shape?(filter, shape_id), do: raise("duplicate shape #{shape_id}")
```
might as well use the new API!